event-sourcing/event-store-postgres/src/main/scala/org/apache/james/eventsourcing/eventstore/postgres/JsonEventSerializer.scala

```diff
@@ -0,0 +1,87 @@
+ /***************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.eventsourcing.eventstore.postgres
+
+import java.io.IOException
+import java.util
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.google.common.collect.ImmutableSet
+import javax.inject.{Inject, Named}
+import org.apache.james.eventsourcing.Event
+import org.apache.james.eventsourcing.eventstore.dto.{EventDTO, EventDTOModule}
+import org.apache.james.json.{DTO, DTOModule, JsonGenericSerializer}
+
+import scala.annotation.varargs
+import scala.jdk.CollectionConverters._
+
+object JsonEventSerializer {
```

Comment:
I think this looks like the same as the one for Cassandra... could be made generic and moved to event-store-api?

Reply:
event-store-api is used for all event store implementations including memory implementation while JsonEventSerializer is only used for Postgres and Cassandra. Should I move it to event-store-api (and other duplicated stuff as well)?

Reply:
Think so. Memory would not use those classes that's it, but I don't see the point of duplicating this code if it's the exact same for cassandra and postgres

Reply:
Huge +1 with @Arsnael remark